be/src/exec/stream/scan/stream_scan_operator.h

```diff
@@ -36,11 +36,58 @@ class StreamScanOperatorFactory final : public ConnectorScanOperatorFactory {
 
 class StreamScanOperator final : public ConnectorScanOperator {
 public:
-    StreamScanOperator(OperatorFactory* factory, int32_t id, int32_t driver_sequence, int32_t dop, ScanNode* scan_node);
+    StreamScanOperator(OperatorFactory* factory, int32_t id, int32_t driver_sequence, int32_t dop, ScanNode* scan_node, bool is_stream = false);
 
     ~StreamScanOperator() override = default;
 
     ChunkSourcePtr create_chunk_source(MorselPtr morsel, int32_t chunk_source_index) override;
+
+    Status reset_epoch(RuntimeState* runtime_state) override {
+        _is_epoch_finished = false;
+        _is_epoch_start = true;
+        _current_epoch_info = _stream_epoch_manager->epoch_info();
+
+        // reset chunkSource status
+        // swap chunk queue
+        int chunk_source_size = _closed_chunk_sources.size();
+        for (int i = 0; i < chunk_source_size ; i++) {
+            ChunkSourcePtr chunk_source_ptr = _closed_chunk_sources.front();
+            _closed_chunk_sources.pop();
+            chunk_source_ptr->reset_status();
+            _old_chunk_sources.push(chunk_source_ptr);
+
+            auto tablet_id = chunk_source_ptr->get_lane_owner();
+            auto binlog_offset = _stream_epoch_manager->get_binlog_offset(runtime_state->fragment_instance_id(), _id, tablet_id);
+            chunk_source_ptr->set_stream_offset(binlog_offset->tablet_version, binlog_offset->lsn);
+            chunk_source_ptr->set_epoch_limit(_current_epoch_info.max_scan_rows, _current_epoch_info.max_exec_millis);
+        }
+
+        return Status::OK();
+    }
+
+    bool is_finished() const override { return _stream_epoch_manager && _stream_epoch_manager->is_finished(); }
+
+    Status set_epoch_finished(RuntimeState* state) override {
```

Comment:
Why not only implement the `set_epoch_finished` at here, instead of the `ScanOperator` ?